app/code/Transoft/Blog/view/frontend/templates/single_blog_post.phtml

```diff
@@ -0,0 +1,30 @@
+<?php
+
+use Magento\Framework\Exception\NoSuchEntityException;
+use Transoft\Blog\Block\SingleBlog;
+?>
+<?php /** @var SingleBlog $block*/ ?>
+<?php
+$id = $block->getBlogId();
+try {
+    $blog = $block->getBlogById($id);
+} catch (NoSuchEntityException $e) {
+    $block->_logger->warning($e);
+}
```

Comment:
Should not be in template. Move to block

Reply:
fixed